docs/how-to/cooperative_groups.rst

```diff
@@ -0,0 +1,468 @@
+
+Grid group
+------------
+
+Represents an inter-block cooperative groups type where the group's participating threads span multiple blocks running the same kernel on the same device. Use the cooperative launch API to enable synchronization across the grid group.
+
+.. code-block:: cpp
+
+  class grid_group;
+
+Constructed via:
+
+.. code-block:: cpp
+
+  grid_group g = this_grid();
+
+The ``thread_rank()`` , ``size()``, ``cg_type()``, ``is_valid()`` and ``sync()`` member functions
+are public of the ``grid_group`` class. For further details, check the :ref:`grid_group references <grid_group_ref>`.
+
+Multi-grid group
+------------------
+
+Represents an inter-device cooperative groups type where the participating threads within the group span multiple devices that run the same kernel on the devices. Use the cooperative launch API to enable synchronization across the multi-grid group.
+
+.. code-block:: cpp
+
+  class multi_grid_group;
+
+Constructed via:
+
+.. code-block:: cpp
+
+  // Kernel must be launched with the cooperative multi-device API
+  multi_grid_group g = this_multi_grid();
+
+The ``num_grids()`` , ``grid_rank()`` , ``thread_rank()``, ``size()``, ``cg_type()``, ``is_valid()`` ,
+and ``sync()`` member functions are public of the ``multi_grid_group`` class. For
+further details check the :ref:`multi_grid_group references <multi_grid_group_ref>` .
+
+Thread-block tile
+------------------
+
+This constructs a templated class derived from ``thread_group``. The template defines the tile
+size of the new thread group at compile time. This group type also supports sub-wave level intrinsics.
+
+.. code-block:: cpp
+
+  template <unsigned int Size, typename ParentT = void>
+  class thread_block_tile;
+
+Constructed via:
+
+.. code-block:: cpp
+
+  template <unsigned int Size, typename ParentT>
+  _CG_QUALIFIER thread_block_tile<Size, ParentT> tiled_partition(const ParentT& g)
+
+
+.. note::
+
+  * Size must be a power of 2 and not larger than warp (wavefront) size.
+  * ``shfl()`` functions support integer or float type.
+
+The ``thread_rank()`` , ``size()``, ``cg_type()``, ``is_valid()``, ``sync()``, ``meta_group_rank()``, ``meta_group_size()``, ``shfl()``, ``shfl_down()``, ``shfl_up()``, ``shfl_xor()``, ``ballot()``, ``any()``, ``all()``, ``match_any()`` and ``match_all()`` member functions are public of the ``thread_block_tile`` class. For further details, check the :ref:`thread_block_tile references <thread_block_tile_ref>` .
+
+Coalesced groups
+------------------
+
+Threads (64 threads on CDNA and 32 threads on RDNA or NVIDIA GPUs) in a warp cannot execute different instructions simultaneously, so conditional branches are executed serially within the warp. When threads encounter a conditional branch, they can diverge, resulting in some threads being disabled, if they do not meet the condition to execute that branch. The active threads referred as coalesced, and coalesced group represents an active thread group within a warp. 
```

Comment:
I wouldn't mention nvidia here and will note that nvidia independent thread scheduling presents the appearance that threads on different branches execute concurrently.  We need a warning here that we do not support independent thread scheduling and some cuda apps using them ported to hip can deadock if they make use of independent thread scheduling.